src/libcollections/binary_heap.rs

```diff
@@ -1194,8 +1194,8 @@ impl<T: Ord, I: IntoIterator<Item = T>> SpecExtend<I> for BinaryHeap<T> {
 }
 
 impl<T: Ord> SpecExtend<BinaryHeap<T>> for BinaryHeap<T> {
-    fn spec_extend(&mut self, ref mut other: BinaryHeap<T>) {
-        self.append(other);
+    fn spec_extend(&mut self, mut other: BinaryHeap<T>) {
```

Comment:
This is purely a coding style issue and both versions are equivalent. Rust doesn't implicitly copy things unless they are Copy, and BinaryHeap is not and we can tell from the trait bounds that T doesn't have to be copy/clone at all to use this method.